src/Core/System.php

```diff
@@ -468,20 +468,27 @@ public static function getLoadAvg(): array
 	 *
 	 * @param string $url  The new Location to redirect
 	 * @param int    $code The redirection code, which is used (Default is 302)
+	 * @param bool   $includeProxyHeaders Whether proxy-related headers are being sent
 	 *
+	 * @return void
 	 * @throws FoundException
 	 * @throws MovedPermanentlyException
 	 * @throws TemporaryRedirectException
-	 *
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function externalRedirect($url, $code = 302)
+	public static function externalRedirect(string $url, int $code = 302, bool $includeProxyHeaders = false)
 	{
 		if (empty(parse_url($url, PHP_URL_SCHEME))) {
 			Logger::warning('No fully qualified URL provided', ['url' => $url, 'callstack' => self::callstack(20)]);
 			DI::baseUrl()->redirect($url);
 		}
 
+		if ($includeProxyHeaders) {
+			// Send Pragma/Cache-Control headers
+			header('Pragma: no-cache');
+			header('Cache-Control: no-cache, must-revalidate');
+		}
```

Comment:
What do these headers have to do with proxies?

Reply:
The team for the squid proxy has explained it in easy but accurate detail:
https://squidproxy.wordpress.com/2012/10/16/squid-3-2-pragma-cache-control-no-cache-versus-storage/

Reply:
Ok, thanks for the link. However, these headers aren't specific to proxies but any HTTP client which Squid is one. This article is informative, including about the `Pragma` header that doesn't make any sense in server responses like the ones we are generating with Friendica, so line 488 can be removed as it is effectively useless. For the other, it sounds like it should be `Cache-Control: no-store` instead. This also removes the redundant `must-revalidate` which is included in `no-cache`.

Since it isn't meant specifically for proxies, the parameter should be renamed to `$noCache` or something in this vein.